Make HUD settings apply consistently

dearmochi opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
Fixes #12943
- HUD alpha, color and style settings should now apply to all HUD elements consistently
- Cleans up some code when possible
- Fixes intent selector using the HUD color setting when it shouldn't

## Why It's Good For The Game
It makes HUD settings actually work instead of for 50% of the elements on the screen, making it more aesthetically pleasing for the player

## Images of changes
![image](https://user-images.githubusercontent.com/1496804/84606244-8ce66380-aea4-11ea-8a87-a154f25982fa.png)
_HUD with Alpha: 200, Color: lightblue, Style: White_

![image](https://user-images.githubusercontent.com/1496804/84606251-a7204180-aea4-11ea-814e-329da8953e11.png)
_The HUD settings do not interfere with the custom action button backgrounds_

## Changelog
:cl:
fix: Fix HUD settings (alpha, color and style) not applying to most of the HUD elements (actions, gear, etc.)
fix: Fix intent selector taking HUD color setting when it shouldn't, causing it to look hard to read
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Fox-McCloud commented:
That big red cross is ugly as hell.

It being greyed out is more than good enough.

BakKeener commented:
I'm with Fox on that, the big red crosses aren't good looking, I thought Greyed out was enough, plus isn't here a text message telling you it doesn't work when it's unavailable when you click it.

Mitchs98 commented:
Echoing on the big red cross. Otherwise it looks nice~

dearmochi commented:
Oof, after sleeping on it I realized it also looks terrible, so I removed the red cross and added a black overlay:
![image](https://user-images.githubusercontent.com/1496804/84629944-6ac40400-aeeb-11ea-8e70-8ee603851e04.png)

Fox-McCloud commented on code/datums/action.dm:
```diff
@@ -126,25 +136,28 @@
+	return TRUE
 
 /datum/action/item_action/ApplyIcon(obj/screen/movable/action_button/current_button)
+	current_button.overlays.Cut()
```
We don't do this anymore.

`cut_overlays`

dearmochi replied:
Found about 170 instances of that, so I replaced them all by `cut_overlays`

Fox-McCloud commented:
Erm, I wouldn't do that in this PR---that'd be better reserved for a PR that exclusively does that. It makes it unceessarily difficult for us to review such things--and it's really beyond the scope of the PR in question. I meant for *That* specific use case and anything related to it.

dearmochi commented:
> Erm, I wouldn't do that in this PR---that'd be better reserved for a PR that exclusively does that. It makes it unceessarily difficult for us to review such things--and it's really beyond the scope of the PR in question. I meant for That specific use case and anything related to it.

Okay, sure, I'll undo that. However I still don't see the point of using the proc you suggested. Is it for readability? Is it a *just in case* like `LAZYLEN` (which you want removed)?

Fox-McCloud commented:
the overlay manipulation system is to give overlays specific priority so that certain overlays with specific priorities can be sorted properly or skipped over--having things outside of it is going to mess up that system.

It's not just a redefine of an existing proc; it's entire its own thing.

When it was added, it wasn't plugged in universally (unfortunately)--but we've been fixing it as we go along/refactor code.

farie82 commented on code/_onclick/hud/action_button.dm:
```diff
@@ -99,14 +99,20 @@
 		icon_state = "bg_alien"
 	else
-		icon_state = "bg_default"
+		icon = user.client ? ui_style2icon(user.client.prefs.UI_style) : initial(icon)
```
Making these two ternaries into an `if/else` would improve readability I think.

farie82 commented on code/_onclick/hud/human.dm:
```diff
@@ -395,9 +402,9 @@
 		if(inv.slot_id)
 			if(inv.slot_id in S.no_equip)
-				inv.alpha = 128
+				inv.alpha = hud_alpha * 0.5
```
`/ 2` can be done here. Divisions and multiplications have mostly the same performance on byond. In some cases divisions are faster than multiplications.
Rule of thumb is to use which one reads the easiest

farie82 commented on code/datums/action.dm:
```diff
@@ -126,25 +136,28 @@
+	return TRUE
 
 /datum/action/item_action/ApplyIcon(obj/screen/movable/action_button/current_button)
+	current_button.cut_overlays()
```
Looking at line 156. It is not even needed to have this here. It'll be cut already there and on line 110 the parent cuts it if you were to go into the else.
That or line 156 can be gone and this statement can be put in the `if(use_itemicon)` scope

dearmochi commented:
@Kyep 

I think there was a misunderstanding on what this PR does. It simply makes the common HUD elements (inventory, hands, modes) use the player's [Custom UI settings](https://user-images.githubusercontent.com/1496804/87282590-b8d12500-c4f4-11ea-9563-1f6c2c90161a.png) in the Game Preferences consistently such that ALL common elements use the same theme rather than only some of them. The player's role(s) has otherwise no bearing on the common elements.

Vampire: 
![image](https://user-images.githubusercontent.com/1496804/87282364-7c9dc480-c4f4-11ea-9f22-62f2f19a515c.png)

Changeling:
![image](https://user-images.githubusercontent.com/1496804/87282435-8d4e3a80-c4f4-11ea-9d6b-fd64b8d91cf6.png)

**As a reference, here how the HUD currently looks on master:**

![image](https://user-images.githubusercontent.com/1496804/87283144-490f6a00-c4f5-11ea-82dc-2b190f2c5abe.png)

Three glaring issues:

1. The intent selector is colored for some reason, making it hard to tell which intent you're actually on.
2. The body part selector *isn't* coloured when it should be.
3. Neither are the **common** action buttons in the top-left corner of the screen.

Issues 2 and 3 result in visual inconsistencies with the HUD. This PR fixes all 3 points

dearmochi commented:
![image](https://user-images.githubusercontent.com/1496804/90985671-bfd06600-e57d-11ea-8afc-ba839da3467d.png)

How it looks now
